crates/matrix-sdk-ui/src/timeline/event_item/content/message.rs

```diff
@@ -79,13 +83,13 @@ impl Message {
             }
         });
 
-        let mut threaded = false;
+        let mut thread_root = None;
         let in_reply_to = c.relates_to.and_then(|relation| match relation {
             message::Relation::Reply { in_reply_to } => {
                 Some(InReplyToDetails::new(in_reply_to.event_id, timeline_items))
             }
             message::Relation::Thread(thread) => {
-                threaded = true;
+                thread_root = Some(thread.event_id);
```

Comment:
Is this always guaranteed to be the thread root? (and not a message in the middle of the thread)

Reply:
Yes, see https://spec.matrix.org/v1.8/client-server-api/#threading.

Reply:
TIL :ok_hand: